src/main/python/systemml/mllearn/keras2caffe.py

```diff
@@ -0,0 +1,489 @@
+from keras.models import load_model
+from keras.models import model_from_json
+from keras.utils.conv_utils import convert_kernel
+
+import numpy as np
+
+
+def load_keras_model(filepath):
+    model = load_model(filepath)
+    return model
+
+
+def load_keras_skeleton_model(filepath):
+    json_file = open(filepath, 'r')
+    model_json = json_file.read()
+    json_file.close()
+    loaded_model = model_from_json(model_json)
+    return loaded_model
+
+
+def load_weights_to_model(model,filepath):
+    model.load_weights(filepath)
+    return model
+
+
+#Currently can only generate a Dense model
+def generate_caffe_model(kModel, filepath, weights_filepath, input_shape=None, phases=None):
+    n = caffe.NetSpec()
+    layers = kModel.layers
+    net_params = dict()
+    input_name = ''
+    label_name = ''
+    for layer in layers:
+        blobs = layer.get_weights()
+
+        if type(layer) == keras.layers.InputLayer:
+            #Grab the batchsize from i 0, shift over channels to index 1, and place the rest into the dictionary
+            #TODO determine when to transform for layer types/input shape
+            num = len(layer.batch_input_shape) - 1 #Range from 1st index to second last
+            #TODO check for image_data_format to be channels_first or channels_last
+            batch_list = [layer.batch_input_shape[0], layer.batch_input_shape[-1]]
+            for i  in range(1 ,num):
+                batch_list.append(layer.batch_input_shape[i])
+            for i in range(len(batch_list)): #Set None dimensions to 0 for Caffe
+                    if( batch_list[i] == None):
+                        batch_list[i] = 1
+                        #print("Batch size of InputLayer: ")
+                        #print(batch_list[i])
+            name = layer.name
+            #TODO figure out having 2 tops, with n.label
+            input_name = name
+            label_name = name + "_label"
+            n[name] = L.Input(shape=[dict(dim=batch_list)])
+
+        elif type(layer) == keras.layers.Dense:
+            #Pull name from Keras
+            name = layer.name
+            #Pull layer name of the layer passing to current layer
+            in_names = getInboundLayers(layer)
+            #Pipe names into caffe using unique Keras layer names
+            n[name] = L.InnerProduct(n[in_names[0].name], num_output=layer.units) #TODO: Assert only 1
+            config = layer.get_config()
+            if config['use_bias']:
+                net_params[name] = (np.array(blobs[0]).transpose(1, 0), np.array(blobs[1]))
+            else:
+                net_params[name] = (blobs[0])
+            if layer.activation is not None and layer.activation.__name__ != 'linear':
+                name_act = name + "_activation_" + layer.activation.__name__ #get function string
+                n[name_act] = getActivation(layer,n[name])
+
+        elif type(layer) == keras.layers.Flatten:
+            
+            """TODO Implement Flatten in Caffe2DML
```

Comment:
Flatten ends up being implicit in Caffe2DML due to the format of storing tensors stretched out into matrices.  Caffe2DML could certainly accept and ignore it, which I believe you are doing here.  In the meantime, could you update this comment to explain how you are ignoring and instead linking the previous layer directly to the next layer?